sql-odbc/src/sqlodbc/opensearch_connection.cpp

```diff
@@ -95,6 +95,15 @@ int LIBOPENSEARCH_connect(ConnectionClass *self) {
     rt_opts.crypt.verify_server = (self->connInfo.verify_server == 1);
     rt_opts.crypt.use_ssl = (self->connInfo.use_ssl == 1);
 
+    // Server host url protocol is necessary for sql plugin validation
+    if (rt_opts.conn.server.size() && rt_opts.conn.server.find("http", 0) != 0) {
```

Comment:
I think you explicitly need to find `http://` OR `https://` with the `://` in the string.

Reply:
This searches from index 0 of the url. `http://` and `https://` will both be caught. If the url has `http` prepended and not have the following `://` or `s://` the SQL plugin query will fail in the same way as any invalid url. Both implementations create the same output but this one avoids another condition. I can change if you prefer explicitly looking for each prefix though!

Reply:
What if the server is `bq-http-opensearch`?

Reply:
Curl command fails when it queries the SQL plugin. If we do have a url like this the downside is that this would produce a confusing url for the user on error. I will revise to search for both `http://` and `https://`.